Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/azure/webapp/deploymentslot/DeploymentSlotNode.java

```diff
@@ -116,149 +123,42 @@ protected void refreshItems() {
         this.renderNode(WebAppBaseState.fromString(slot.state()));
     }
 
-    // Delete action class
-    private class DeleteAction extends NodeActionListener implements ActionBackgroundable, ActionPromptable, ActionTelemetrable {
-
-        @Override
-        protected void actionPerformed(NodeActionEvent e) {
-            DeploymentSlotNode.this.getParent()
-                    .removeNode(DeploymentSlotNode.this.getSubscriptionId(), DeploymentSlotNode.this.getName(), DeploymentSlotNode.this);
-        }
-
-        @Override
-        public String getPromptMessage() {
-            return Node.getPromptMessage(AzureActionEnum.DELETE.getName().toLowerCase(), DeploymentSlotModule.MODULE_NAME, DeploymentSlotNode.this.name);
-        }
-
-        @Override
-        public String getProgressMessage() {
-            return Node.getProgressMessage(AzureActionEnum.DELETE.getDoingName(), DeploymentSlotModule.MODULE_NAME, DeploymentSlotNode.this.name);
-        }
-
-        @Override
-        public TelemetryParameter getTelemetryParameter() {
-            return TelemetryParameter.WebApp.DeploymentSlot.DELETE;
-        }
+    @AzureOperation(value = "start deployment slot", type = AzureOperation.Type.ACTION)
+    private void start() {
+        AzureWebAppMvpModel.getInstance().startDeploymentSlot(subscriptionId, webAppId, slotName);
```

Comment:
In function we get the status after start/stop/restart but deployment slot just render the action directly, shall we unify the behaviors?

Reply:
Will unity them including WebApp in a individual PR later.